coderd/healthcheck/health/model.go

```diff
@@ -0,0 +1,20 @@
+package health
+
+const (
+	SeverityOK      Severity = "ok"
+	SeverityWarning Severity = "warning"
+	SeverityError   Severity = "error"
+)
+
+// @typescript-generate Severity
+type Severity string
+
+var severityRank = map[Severity]int{
+	SeverityOK:      0,
+	SeverityWarning: 1,
+	SeverityError:   2,
+}
+
+func (s Severity) Value() int {
+	return severityRank[s]
+}
```

Comment:
My immediate preference would be for Severity to be an int value that implements Stringer so that you can natively do `if s1 > s2 { ... }` but I'm guessing that won't play nicely with our typescript generator?

Reply:
I see DevOps people setting up scripts this way: `curl https://coder.tld/health | grep warning` or `curl https://coder.tld/health | grep error`, hence I primarily focused on string. I will check the generator 👍

Reply:
That's fair, and I'd still prefer for it to be emitted as a string value in our JSON payloads. Just having it defined internally as an integer has some nice properties.

Reply:
Ok, I tried to implement this, but failed sooner than expected. I had to implement UnmarshalJSON and MarshalJSON, but `swaggo` ignore these methods and [fails](https://github.com/swaggo/swag/issues/1367) with:

`2023/11/23 15:50:52 ParseComment error in file /Users/mtojek/code/coder/coderd/debug.go :enum value ok can't convert to integer err: strconv.Atoi: parsing "ok": invalid syntax` 

so I removed samples: `enums:"ok,warning,error"`, and retried. `make gen` passed, but as expected TS can only see this field as `number`:

```
2875     severity: "ok",
         ~~~~~~~~

  src/api/typesGenerated.ts:2124:12
    2124   readonly severity: HealthSeverity;
                    ~~~~~~~~
    The expected type comes from property 'severity' which is declared here on type 'HealthcheckWebsocketReport'
```

I think I will leave it as is, there is too much hacking around to make working properly.